manager/vm/vm.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Ultraviolet
+// SPDX-License-Identifier: Apache-2.0
+package vm
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/gofrs/uuid"
+	"github.com/ultravioletrs/cocos/agent"
+	"github.com/ultravioletrs/cocos/internal"
+	"github.com/ultravioletrs/cocos/manager/qemu"
+	"github.com/ultravioletrs/cocos/pkg/manager"
+)
+
+const (
+	firmwareVars = "OVMF_VARS"
+	KernelFile   = "bzImage"
+	rootfsFile   = "rootfs.cpio"
+)
+
+// VM represents a virtual machine.
+//
+//go:generate mockery --name VM --output=./mocks --filename vm.go --quiet --note "Copyright (c) Ultraviolet \n // SPDX-License-Identifier: Apache-2.0"
+type VM interface {
+	Start() error
+	Stop() error
+	SendAgentConfig(ac agent.Computation) error
+}
+
+type vm struct {
+	config        qemu.Config
+	cmd           *exec.Cmd
+	logsChan      chan *manager.ClientStreamMessage
+	computationId string
+}
+
+//go:generate mockery --name VMFactory --output=./mocks --filename vm_factory.go --quiet --note "Copyright (c) Ultraviolet \n // SPDX-License-Identifier: Apache-2.0"
+type VMFactory func(config qemu.Config, logsChan chan *manager.ClientStreamMessage, computationId string) VM
```

Comment:
Let's name it to `VMProvider` or, even better, just `Provider` so we have `vm.Provider`. Also, move the implementation of the VM to `qemu` since this only one pmplementation and we are going to have others soon.